flocker/node/agents/cinder.py

```diff
@@ -449,13 +449,14 @@ def compute_instance_id(self):
 
         # If we've got this correct there should only be one matching instance.
         # But we don't currently test this directly. See FLOC-2281.
-        if len(matching_instances) == 1:
+        if len(matching_instances) == 1 and matching_instances[0]:
             return matching_instances[0]
         # If there was no match, or if multiple matches were found, log an
         # error containing all the local and remote IPs.
         COMPUTE_INSTANCE_ID_NOT_FOUND(
             local_ips=local_ips, api_ips=api_ip_map
         ).write()
+        raise UnknownInstanceID(self)
```

Comment:
It would be pretty nice to have automated test coverage for this case, particularly if we anticipate ever meaningfully handling `UnknownInstanceID` in the deployer.  A verified fake of nova might make it possible to unit test this (I hear there is one, I don't think we use it anywhere for anything yet).